src/msm.rs

```diff
@@ -133,12 +133,17 @@ impl VariableBaseMSM {
         window_bits: u32,
         max_batch: u32,
         max_collisions: u32,
+        glv_enabled: bool,
     ) -> GroupProjective<P> {
         assert!(
             window_bits as usize > GROUP_SIZE_IN_BITS,
             "Window_bits must be greater than the default log(group size)"
         );
-        if TypeId::of::<P>() == TypeId::of::<G1Parameters>() {
+        if glv_enabled {
+            assert!(
+                TypeId::of::<P>() == TypeId::of::<G1Parameters>(),
+                "Glv is only supported for bls12-381 curve!"
+            );
```

Comment:
I'll try to express this in terms of traits so that there's no need for such an `if`, as not the best option through the TypeId check now began to require even more attention from the caller